### DISCUSSION TITLE
Add companion object to existing class

### DISCUSSION THREAD

**Discussion by VapidLinus** [square/kotlinpoet#1245] [Q&A] [answered]
Is it possible to add a companion object to an existing class - i.e. a class that was written manually and not with KotlinPoet?

I've seen in the docs that it's possible to add a companion object to a generated class but it doesn't mention if it's possible to attach to an existing one.

kotlinx.serialization has some way of adding generating a companion object with its `@Serializable` attribute and I want to achieve something similar.

**Comment by ZacSweers** [comment, marked as answer]
No, kotlinpoet is for generating new source files, not IR transformations (which is what kotlinx-serialization leverages).

**Reply by VapidLinus** [reply]
Thanks for the answer and for pointing me in the right direction. I'll look into IR transformations as a solution to my approach!
